src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -259,6 +259,26 @@ internal static void CheckUnsafeModifier(this Symbol symbol, DeclarationModifier
             }
         }
 
+        /// <summary>
+        /// Does the top level type containing this symbol have 'Microsoft.CodeAnalysis.Embedded' attribute?
+        /// </summary>
+        public static bool IsHiddenByCodeAnalysisEmbeddedAttribute(this Symbol symbol)
+        {
+            // Only upper-level types should be checked 
+            var upperLevelType = symbol.Kind == SymbolKind.NamedType ? (NamedTypeSymbol)symbol : symbol.ContainingType;
+            if ((object)upperLevelType == null)
```

Comment:
This is called by things that have no upper level type as well, like namespaces. It can be called by any symbol.

---
In reply to: [113549150](https://github.com/dotnet/roslyn/pull/18715#discussion_r113549150) [](ancestors = 113549150)